### ISSUE TITLE
Issue with mySQL date column and JSONATA $spread() function

### ISSUE THREAD

**Issue by rakgupta** [node-red/node-red-nodes#880]
### Which node are you reporting an issue on?
mySQL node (v 1.0.0)
### What are the steps to reproduce?
Create a select statement with at least 2 columns from a mySQL database (one should be a date/time column) [e.g. myCol, created]. Connect a change node to the output of the mySQL node and use the JSONATA $spread() function to parse the columns (e.g. msg.myCol = $spread(payload.myCol), msg.created = $spread(payload.created)). Connect a debug node to the change node.
### What happens?
The JSONATA function will parse the non-date column correctly and create an array with all the "myCol" values from the payload. The date column (created) will not be parsed and return a array with zero elements.

If you copy the payload from the debug panel and send it through an inject node, it works as expected.

**NOTE:** The link to the details are in this thread on the Node-RED forum:
https://discourse.nodered.org/t/possible-issue-with-mysql-date-column-and-jsonata-spread-function/57257

### What do you expect to happen?
Both columns should be parsed correctly into their own arrays.

### Please tell us about your environment:

- [ ] Node-RED version:  v 1.3.5
- [ ] mySQL node version: 1.0.0
- [ ] node.js version: 14.15.5
- [ ] npm version:
- [ ] Platform/OS: MacOS
- [ ] Browser:Safari

**Comment by dceejay** [comment]
From the discussion it looks like this is more a jsonata things than a mysql node bug ?

**Comment by rakgupta** [comment]
> more a jsonata things than a mysql node bug

Possibly - I wasn't sure. However, if I take the values from the output from the mySQL node and copy them into an inject node, it works correctly so thought I would create the issue. My thought is that it may have something to do with a date object (returned by mySQL) vs text (possibly that's what the inject node does to the values)?

**Comment by rakgupta** [comment]
> From the discussion it looks like this is more a jsonata things than a mysql node bug ?

I did a test based on one of the solutions that was proposed in the forum. It generates data similar to what is returned in the SQL query (including the date in the same format). I put this through the $spread function and it worked correctly, so it's probably not a JSONATA issue? Below is the flow that I used to test.

```
[{"id":"337250983f7905d8","type":"inject","z":"9dd3cdef.2f2f38","name":"","props":[{"p":"payload"},{"p":"props","v":"[\"totalHomeuse\",\"gridToHome\",\"solarToHome\",\"batteryToHome\",\"totalSolar\",\"solarToGrid\",\"solarToBattery\",\"day\"]","vt":"json"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":279.6666564941406,"y":171.99990844726562,"wires":[["da751414a159fccd"]]},{"id":"da751414a159fccd","type":"function","z":"9dd3cdef.2f2f38","name":"generate dummy data (ignore me)","func":"\nconst templateRow = {\n    totalHomeuse: 67.74,\n    gridToHome: 34.62,\n    solarToHome: 21.11,\n    batteryToHome: 12.01,\n    totalSolar: 34.88,\n    solarToGrid: 0.05,\n    solarToBattery: 13.72,\n    day: \"2022-01-17T06:00:00.0002\"\n}\n\nconst data = []\nfor (let index = 0; index < 10; index++) {\n    const element = RED.util.cloneMessage(templateRow);\n    element.totalHomeuse = randomFloat(60,70)\n    element.gridToHome = randomFloat(30,40)\n    element.solarToHome = randomFloat(20,30)\n    element.batteryToHome = randomFloat(10,20)\n    element.totalSolar = randomFloat(30,40)\n    element.solarToGrid = randomFloat(0.01,0.99)\n    element.solarToBattery = randomFloat(11,14)\n    element.day = new Date(Date.now() + ((index+1)*1000*60)).toISOString()\n    data.push(element)\n}\nmsg.payload = data;\n\n\nreturn msg;\n\n\n/**\n* @param {number} min\n* @param {number} max\n*/\nfunction randomFloat(min, max) {\n    return Math.random() * (max - min) + min;\n};","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":517.6666564941406,"y":171.99990844726562,"wires":[["942662580d53e493","e3e67cb9207e5e23","2f1ca195.87b276"]]},{"id":"942662580d53e493","type":"debug","z":"9dd3cdef.2f2f38","name":"","active":true,"tosidebar":true,"console":false,"tostatus":true,"complete":"payload","targetType":"msg","statusVal":"payload","statusType":"auto","x":769.6666564941406,"y":171.99990844726562,"wires":[]},{"id":"2f1ca195.87b276","type":"change","z":"9dd3cdef.2f2f38","name":"Spread","rules":[{"t":"set","p":"totalHomeuse","pt":"msg","to":"$spread(payload.totalHomeuse)","tot":"jsonata"},{"t":"set","p":"dateTime","pt":"msg","to":"$spread(payload.day)","tot":"jsonata"}],"action":"","property":"","from":"","to":"","reg":false,"x":833.3333435058594,"y":82.00000762939453,"wires":[["e8504ec0.fc0998"]]},{"id":"e8504ec0.fc0998","type":"debug","z":"9dd3cdef.2f2f38","name":"After Spread","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":1048.3333435058594,"y":82.00000762939453,"wires":[]}]
```

**Comment by dceejay** [comment]
can you add some debug to the output of the mysql node  - maybe a function that prints out the typeof of the day properties - so check exactly what type it is. 

The short answer is we probably won't change anything in the node as others may already be relying on the fact it returns in the format it does already - and it may just be a case of using a function with array.map or somesuch to change the type before you process it.

**Comment by rakgupta** [comment]
> can you add some debug to the output of the mysql node
I added a "typeof" and debug says it is an "object"
<img width="358" alt="Screen Shot 2022-01-29 at 7 50 12 AM" src="https://user-images.githubusercontent.com/59174606/151663588-41cbd202-2cc0-4ba9-a9bc-cff2cc8da0fe.png">

> The short answer is we probably won't change anything in the node as others may already be relying on the fact it returns in the format it does already
I think I may be one of those people - I remember sometime last year when the date type was changed unintentionally (I think) and it broke my flows. You fixed it right away and it's been working just fine. I have a work around, so I am going to close the issue.

Thanks again for your quick response.
